5 pattern for java workload monitoring

freschri opened the pull request:
**Note: this PR depends on
https://github.com/aws-observability/aws-observability-accelerator/pull/15 and on
https://github.com/aws-quickstart/cdk-eks-blueprints/pull/784**
some changes in here are also in https://github.com/aws-observability/cdk-aws-observability-accelerator/pull/65

shapirov103 commented on cdk.json:
```diff
@@ -17,12 +17,23 @@
+      },
+      "fluxTargetNamespace": "grafana-operator",
+      "bootstrapValues": {
+        "GRAFANA_CLUSTER_DASH_URL" : "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/cluster.json",
```
Is there a reason why this is an FQDN URL which happens to point to the same repo as the bootstrap repo? Is there a way to use the dashboard location based off the relative path in the repo and branch that is referenced here?

freschri replied:
urls are copied from the existing ones, no change, I guess you are asking for an improvement?
I dont think it is possible anyway, because the urls are used in the GrafanaDashboard manifest for the grafana operator, e.g. see https://github.com/freschri/aws-observability-accelerator/blob/java-pattern/artifacts/grafana-operator-manifests/eks/java/amg_grafana-dashboards.yaml

shapirov103 commented on cdk.json:
```diff
@@ -17,12 +17,23 @@
-    "nodeexporter.dashboard.url": "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/nodeexporter-nodes.json",
-    "nodes.dashboard.url": "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/nodes.json",
-    "workloads.dashboard.url": "https://raw.githubusercontent.com/aws-observability/aws-observability-accelerator/main/artifacts/grafana-dashboards/eks/infrastructure/workloads.json",
+    "grafanaGitOpsConfig": {
```
What is the reason for this change? Just general refactoring to reduce code duplication?

freschri replied:
basically yes: before this change, if I needed to e.g. like in this case change the flux path, I needed to duplicate the index file into another pattern and edit the single line..

elamaran11 replied:
I like this approach.

shapirov103 reviewed (commented):
@freschri great work,  I added a few questions/comments. 
I understand this has dependency so can't be merged yet.

freschri commented on lib/common/addons/amp-rules-configurator/amp-rules-configurator-addon.ts:
```diff
@@ -0,0 +1,54 @@
+/**
+ * Implementation of Rules configuration for AMP
+ */
+export class AmpRulesConfiguratorAddOn implements blueprints.ClusterAddOn {
```
please see reply here: https://github.com/aws-observability/cdk-aws-observability-accelerator/pull/65#discussion_r1269145346

freschri commented on lib/common/addons/amp-rules-configurator/amp-rules-configurator-addon.ts:
```diff
@@ -0,0 +1,54 @@
+export class AmpRulesConfiguratorAddOn implements blueprints.ClusterAddOn {
+
+    readonly ampRulesConfiguratorAddOn: AmpRulesConfiguratorAddOnProps;
+    private _ruleGroupsNamespaces: CfnRuleGroupsNamespace[] = [];
```
done

elamaran11 commented on docs/patterns/single-new-eks-observability-accelerators/single-new-eks-java-opensource-observability.md:
```diff
@@ -0,0 +1,124 @@
+# Single New EKS Cluster Open Source Observability Accelerator - monitoring Java applications
```
Where is the pattern code for this doc?

freschri replied:
small code changes to support new java flag and cdk file config. main code changes are in otel collector config (see quickstart PR).

elamaran11 reviewed (commented):
@freschri I love the PR, great work for sure. I appreciate the fact that you have also added approach to refactor the context. I have following feedback :
1. I dont see a real pattern for Java workload monitoring. All i see is a doc showing the approach. We want a real pattern for our customers to try in single click and thats the purpose of these accelerators. I propose two approaches: first is creating a flag in context which says Java_Mon: Yes which will also apply java mon to `existing` and `single-new` open source patterns or second approach is to create all new pattern for existing or new single java monitoring for Open source so people will have clarity.
2. why do we see files for alerting-rules and recording-rules in this PR?

elamaran11 reviewed (changes requested):
@freschri I love the PR, great work for sure. I appreciate the fact that you have also added approach to refactor the context. I have following feedback :

I dont see a real pattern for Java workload monitoring. All i see is a doc showing the approach. We want a real pattern for our customers to try in single click and thats the purpose of these accelerators. I propose two approaches: first is creating a flag in context which says Java_Mon: Yes which will also apply java mon to existing and single-new open source patterns or second approach is to create all new pattern for existing or new single java monitoring for Open source so people will have clarity.
why do we see files for alerting-rules and recording-rules in this PR?

Also fix the GH Warnings

elamaran11 commented:
@freschri Do we have an update on this PR?

freschri commented:
@elamaran11 I added a flag to enable Java observability. Updated doc too. 2 dependencies on quickstart PRs:
https://github.com/aws-quickstart/cdk-eks-blueprints/pull/796
https://github.com/aws-quickstart/cdk-eks-blueprints/pull/776

freschri commented:
> @freschri Do we have an update on this PR?

@elamaran11 I added a flag to enable Java observability. Updated doc too. 2 dependencies on quickstart PRs:
https://github.com/aws-quickstart/cdk-eks-blueprints/pull/796
https://github.com/aws-quickstart/cdk-eks-blueprints/pull/776

freschri commented:
> @freschri I love the PR, great work for sure. I appreciate the fact that you have also added approach to refactor the context. I have following feedback :
> 
>     1. I dont see a real pattern for Java workload monitoring. All i see is a doc showing the approach. We want a real pattern for our customers to try in single click and thats the purpose of these accelerators. I propose two approaches: first is creating a flag in context which says Java_Mon: Yes which will also apply java mon to `existing` and `single-new` open source patterns or second approach is to create all new pattern for existing or new single java monitoring for Open source so people will have clarity.
> 
>     2. why do we see files for alerting-rules and recording-rules in this PR?

Because there are dashboards in the pattern which need fixing. recording rules fix the the dashboards.

elamaran11 commented:
@freschri Please fix the GH Action issues.

elamaran11 reviewed (changes requested):
@freschri This is AWSome work, thankyou for doing this. This a great new feature addition. I think approach is good, have some minor tactical feedback to fix the gaps :
1. Please merge from main, some of the changes like version, blueprints version is addressed.
2. AMP Alerting and recording rules are missing for Java Mon please AMP Alerting Rules and Recording rules are missing for Java Monitoring. Check [Terraform](https://github.com/aws-observability/terraform-aws-observability-accelerator/blob/main/modules/eks-monitoring/patterns/java/main.tf) for reference terraform reference.
3. A Separate documentation is needed in existing pattern too for Java monitoring.
4. Java mon code should be added to Graviton OSS pattern and exising OSS pattern

freschri commented on docs/patterns/single-new-eks-observability-accelerators/single-new-eks-java-opensource-observability.md:
````diff
@@ -0,0 +1,134 @@
+  }
+```
+
+Please also replace the `AmpAddOn` initialization in the `index.ts` file with:
````
yes thanks updated

freschri commented on lib/single-new-eks-awsnative-observability-construct/index.ts:
```diff
@@ -25,6 +25,7 @@ export default class SingleNewEksClusterAWSNativeobservabilityConstruct {
         ObservabilityBuilder.builder()
             .account(account)
             .region(region)
+            .version("auto")
```
done

freschri commented on lib/single-new-eks-cluster-construct/index.ts:
```diff
@@ -17,6 +17,7 @@ export default class SingleNewEksConstruct {
         ObservabilityBuilder.builder()
             .account(account)
             .region(region)
+            .version("auto")
```
done

freschri commented on lib/single-new-eks-mixed-observability-construct/index.ts:
```diff
@@ -34,6 +34,7 @@ export default class SingleNewEksMixedobservabilityConstruct {
         ObservabilityBuilder.builder()
             .account(account)
             .region(region)
+            .version("auto")
```
done

freschri commented on lib/single-new-eks-opensource-observability-construct/index.ts:
```diff
@@ -14,20 +13,40 @@ export default class SingleNewEksOpenSourceobservabilityConstruct {
 
+        if (utils.valueFromContext(scope, "java.pattern.enabled", false)) {
+            ampAddOnProps.openTelemetryCollector = {
+                manifestPath: __dirname + '/../common/resources/otel-collector-config.yml',
```
yes thanks changed

freschri commented on lib/single-new-eks-opensource-observability-construct/graviton-index.ts:
```diff
@@ -38,33 +37,21 @@ export default class SingleNewEksGravitonOpenSourceObservabilityConstruct {
+                        __dirname + '/../common/resources/amp-config/alerting-rules.yml',
+                        __dirname + '/../common/resources/amp-config/recording-rules.yml'
+                    ]
+                }
```
done, doc updated too

freschri commented on lib/existing-eks-opensource-observability-construct/index.ts:
```diff
@@ -54,33 +53,21 @@ export default class ExistingEksOpenSourceobservabilityConstruct {
-                    "GRAFANA_WORKLOADS_DASH_URL" : workloadsDashUrl
-                },
-            }),
+            new blueprints.addons.FluxCDAddOn({"repositories": [fluxRepository]}),
```
done

freschri commented on lib/existing-eks-opensource-observability-construct/index.ts:
```diff
@@ -54,33 +53,21 @@ export default class ExistingEksOpenSourceobservabilityConstruct {
+                        __dirname + '/../common/resources/amp-config/recording-rules.yml'
+                    ]
+                }
             }),
```
done

elamaran11 commented on lib/existing-eks-opensource-observability-pattern/index.ts:
```diff
@@ -39,12 +40,35 @@ export default class ExistingEksOpenSourceobservabilityPattern {
+            }
+        };
+
+        if (utils.valueFromContext(scope, "java.pattern.enabled", false)) {
```
`(utils.valueFromContext(scope, "java.pattern.enabled", undefined)` situation is not handled. I think people may ignore this in few patterns.

freschri replied:
false is the default value is not found. I guess no change here right?

elamaran11 replied:
I think i agree.

elamaran11 commented on lib/single-new-eks-opensource-observability-pattern/graviton-index.ts:
```diff
@@ -15,17 +15,41 @@ export default class SingleNewEksGravitonOpenSourceObservabilityPattern {
+            }
+        };
+
+        if (utils.valueFromContext(scope, "java.pattern.enabled", false)) {
```
(utils.valueFromContext(scope, "java.pattern.enabled", undefined) situation is not handled. I think people may ignore this in few patterns.

freschri replied:
same as above

elamaran11 commented on lib/single-new-eks-opensource-observability-pattern/index.ts:
```diff
@@ -14,20 +14,44 @@ export default class SingleNewEksOpenSourceobservabilityPattern {
+            }
+        };
+
+        if (utils.valueFromContext(scope, "java.pattern.enabled", false)) {
```
(utils.valueFromContext(scope, "java.pattern.enabled", undefined) situation is not handled. I think people may ignore this in few patterns.

freschri replied:
same as above

elamaran11 reviewed (commented):
@freschri Everything looks great, just minor error handling is missed. This is fixing so many issues btw. Im currently testing it locally, will let you know the outcome. Also will run the e2e here.

elamaran11 commented:
/do-e2e-test single-new-eks-opensource-observability deploy

elamaran11 commented:
@freschri I was able to run this locally and i can see all Dashboards showing data, great work. Once you fix the above issues and resolve both the commands, we can proceed to merge.

elamaran11 commented:
/do-e2e-test single-new-eks-opensource-observability deploy

elamaran11 commented:
/do-e2e-test single-new-eks-opensource-observability deploy

elamaran11 commented:
/do-e2e-test single-new-eks-opensource-observability destroy

elamaran11 commented:
@freschri Please merge from Main, some work from upstream is missing. Once done i can merge.

elamaran11 reviewed (approved):
LGTM. Nice work @freschri
